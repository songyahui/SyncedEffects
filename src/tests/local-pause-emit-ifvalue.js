"use hopscript"

const hh = require("hiphop");


const prg =
      <hh.module>
	<hh.loop>
	  <hh.local L>
	    <hh.pause/>
	    <hh.emit L/>
	    <hh.if apply=${function(){return ! this.L.nowval}}>
	      <hh.atom apply=${function(){console.log("L:", this.L.nowval)}}/>
	    </hh.if>
	  </hh.local>
	</hh.loop>
      </hh.module>;

const m = new hh.ReactiveMachine(prg)

m.react();
m.react();m.react();m.react();
