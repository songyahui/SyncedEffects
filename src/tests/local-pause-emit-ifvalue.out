L: undefined
L: undefined
L: undefined
